src/Core/Notifications/INotificationService.cs

```diff
@@ -60,24 +62,10 @@ public void ShowNotification(INotification notification)
 
             threadHandling.RunOnUIThread(() =>
             {
-                try
+                lock (Locker)
                 {
                     RemoveExistingInfoBar();
-
-                    var buttonTexts = notification.Actions.Select(x => x.CommandText).ToArray();
-
-                    var infoBar = infoBarManager.AttachInfoBarWithButtons(ErrorListToolWindowGuid,
-                        notification.Message,
-                        buttonTexts,
-                        SonarLintImageMoniker.OfficialSonarLintMoniker);
-
-                    activeNotification = new Tuple<IInfoBar, INotification>(infoBar, notification);
-                    activeNotification.Item1.ButtonClick += CurrentInfoBar_ButtonClick;
-                    activeNotification.Item1.Closed += CurrentInfoBar_Closed;
-                }
-                catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
```

Comment:
Minor change: the catch used to cover both RemoveExisting and showing the new, but there is a try-catch inside Remove. So I changed the scope to cover Show.